### ISSUE TITLE
Test so that the hash table returned from Get-TargetResource contain all properties in the schema.mof

### ISSUE THREAD

**Issue by johlju** [PowerShell/DscResource.Tests#220]
Today there are resource that do not return all properties in the hash table that are returned from `Get-TargetResource`. I suggest we have a test that the repository must opt-in for it fail if not all properties are returned. The test should always return the result but not fail unless opt-in. 

The below extract is from the the documentation.

>In the Get-TargetResource ... This function must return a hash table that lists all the resource properties as keys and the actual values of these properties as the corresponding values.
>https://docs.microsoft.com/en-us/powershell/dsc/authoringresourcemof#writing-the-resource-script

This issue has been moved from https://github.com/PowerShell/DscResources/issues/318 and more information exist in that issue.

**Comment by PlagueHO** [comment]
Is it worth considering always running the test and reporting the output but only failing if the opt-in is set? I definitely see value in this test.

**Comment by johlju** [comment]
I agree. Actually meant for the issue description to say that, but it didn’t, but now it does. Changed the issue description. 😄
